### ISSUE TITLE
MIssing validation of channel parameters

### ISSUE THREAD

**Issue by jdall** [asyncapi/parser-js#24]
It would be nice if the parser could validate channel parameters
Having e.g.
```
channels
  playground/testParams/{param1}
```   
without the parameters being specified like e.g.
```
    parameters:
      - name: param1
        schema:
          type: string
```
should result in the parser bailing out - with a useful error message.
Please consider this.
Thanks.

**Comment by fmvilas** [comment]
Absolutely agree. Side note: take into account that new syntax for parameters is going to be like this:

```yaml
channels:
  playground/testParams/{param1}:
    parameters:
      param1:
        schema:
          type: string
```

Parameters is now a map instead of an array.

**Comment by jdall** [comment]
Thanks, @fmvilas .

What's the ETA of this new syntax?

And just as a side note: I cannot locate a description of the schema.type element in the asyncapi docs

**Comment by fmvilas** [comment]
It's already out: https://github.com/asyncapi/asyncapi/releases/tag/2.0.0-rc2.

You can't find Schema type because Schema is a superset of JSON Schema Draft 07: https://github.com/asyncapi/asyncapi/blob/master/versions/next/asyncapi.md#properties. Refer to JSON Schema Draft 07 docs instead.

**Comment by jdall** [comment]
Got it, @fmvilas.

Thank you for you informative comments.

I understand that the Schema is defined elsewhere, but shouldn't the list of properties of the schema object at least mention the 'type' and other mandatory properties. Not having it in there  makes it a little less straight forward to grasp the spec.

**Comment by fmvilas** [comment]
What do you mean by "mention"? It's mentioned there (the second on the list) and there's a link to JSON Schema specification one paragraph above: https://github.com/asyncapi/asyncapi/blob/master/versions/next/asyncapi.md#schema-object.

It says:
> Further information about the properties can be found in [JSON Schema Core](https://tools.ietf.org/html/draft-handrews-json-schema-01) and [JSON Schema Validation](https://tools.ietf.org/html/draft-handrews-json-schema-validation-01).

Keep in mind this is AsyncAPI **specification**. It's not meant to be a getting started guide or a tutorial. The specification must remain formal, and we can't define things that are not defined by the spec.

I agree this could be easier to grasp but that's what guides and tutorials are for. I'll work on this after version 2.0.0 is released next week.

**Comment by jdall** [comment]
I understand your confusion. My bad.

I hadn't realised that rc2 docs were in fact available. The asyncapi home page only references rc1, and that page has been my entry point to the specs so far.

Please consider updating the home page to reflect the availability of the rc2 spec.

**Comment by fmvilas** [comment]
Yes, I have to make it clear that the website is not the source of truth. It may have been a bad idea to put the spec there. Thanks and sorry for creating this confusion.

**Comment by jdall** [comment]
No worries.
You and others are doing a great job in developing asyncapi.
Working with bleeding edge, there are bound to be some hickups here and there, but with the level of support you provide, they are not that big a deal.

Thank you for making it possible to specify broker based APIs in such a great way!

**Comment by fmvilas** [comment]
Thanks for your kind words ☺️

**Comment by derberg** [comment]
@fmvilas this sounds to me like a pretty "manual" validation not handlable by schema. Same is with parameters/variables in server url. Where would you put it in the parser? is it the intention of  `iterateDocument` https://github.com/asyncapi/parser-js/blob/master/lib/parser.js#L150 function to trigger all those additional custom validations, or are there no strict rules?

**Comment by fmvilas** [comment]
No strict rules but I'll trigger these validations right after we do JSON Schema validation with `ajv`, so `iterateDocument` sounds like a good candidate for that. Although, I'd change the name to something more descriptive.

**Comment by derberg** [comment]
Hey folks, I've just opened a PR, it is still missing few things but main validation logic is in place.

I have once concern @fmvila, I think we discussed someday error levels but I can't recall what was the outcome of the discussion. My concern is that this validation is an error now, but it feels like a warning, I mean that this kind of error should not break the functionality of tools like for example generator. My question is what is the vision of the parser, should we just in the generator give user an option to ignore validation errors?
